pkg/router/template/router.go

```diff
@@ -539,8 +539,12 @@ func (r *templateRouter) writeCertificates(cfg *ServiceAliasConfig) error {
 }
 
 // reloadRouter executes the router's reload script.
-func (r *templateRouter) reloadRouter() error {
+func (r *templateRouter) reloadRouter(shutdown bool) error {
 	cmd := exec.Command(r.reloadScriptPath)
+	if shutdown {
+		cmd.Env = append([]string{}, os.Environ()...)
+		cmd.Env = append(cmd.Env, "ROUTER_SHUTDOWN=true")
```

Comment:
I guess you are trying to avoid mutating `os.Environ()`?  The [godoc for `os.Environ`](https://golang.org/pkg/os/#Environ) says it "returns a copy of strings representing the environment", so it should be safe to do the following:
```go
		cmd.Env = append(os.Environ(), "ROUTER_SHUTDOWN=true")
```
I don't object to what you have though, I just found it a little confusing.

Reply:
So it does, thanks.